### PR TITLE
feat(reconnect): add duration metrics

### LINKED ISSUE #11431: A new metric for reconnect duration
There is a dashboard where all "reconnect started" and "reconnect finished" events are shown both on the teacher and on the learner side. In addition to that it would be great to have another metric about reconnect durations, and then add a new panel to the same dashboard to visualize this metric.

### PR DESCRIPTION
**Pull request by anthony-swirldslabs**
**Description**:
1. In `ReconnectMetrics.java`, I'm adding support for measuring the reconnect duration and emitting the duration metrics upon calls to corresponding `increment*Times()` methods.
2. In `reconnect.json` I'm adding two additional panels to the Reconnect dashboard to expose the new metrics in Grafana.

Please note that it's out of scope of this PR to expose any more advanced metrics, such as the total duration of all reconnect attempts before a node finally becomes active. This may (or may not) be implemented in the future in a separate issue/PR.

**Related issue(s)**:

Fixes #11431

**Notes for reviewer**:
`gr assemble` passes.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by cody-littley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java [hunk 1]
```diff
@@ -61,6 +63,27 @@ public class ReconnectMetrics {
      */
     private final Map<NodeId, CountPerSecond> rejectionFrequency = new HashMap<>();
 
+    private static final LongAccumulator.Config SENDER_DURATION_CONFIG = new LongAccumulator.Config(
+                    RECONNECT_CATEGORY, "senderReconnectDurationMs")
+            .withInitialValue(0)
+            .withAccumulator(Long::sum)
+            .withDescription("duration of reconnect as a sender, ms");
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java:70]: please separate (or add) the `ms` to the `.withUnit()` builder method.

Comment (anthony-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java]: Good idea. Just added. Please have a look.

**Review by artemananiev — APPROVED** [review]

**Review by nathanklick — DISMISSED** [review]
LGTM but please have @beeradb also look at the dashboard changes.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java [hunk 2]
```diff
@@ -61,6 +64,29 @@ public class ReconnectMetrics {
      */
     private final Map<NodeId, CountPerSecond> rejectionFrequency = new HashMap<>();
 
+    private static final LongAccumulator.Config SENDER_DURATION_CONFIG = new LongAccumulator.Config(
+                    RECONNECT_CATEGORY, "senderReconnectDurationMs")
```
[the remark this answers is not in the document]
Comment (anthony-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java]: Sounds good. I've replaced milliseconds with seconds. Please have a look.

**Review by kimbor — APPROVED** [review]

**Review by beeradb — APPROVED** [review]
The metric and dashboard changes LGTM on the surface.

We currently don't have a good process for running these pre-release builds to verify these dashboard/metric name changes. Perhaps moving forward we could do before/after screenshots showing the changes?

**Review by Neeharika-Sompalli — APPROVED** [review]

**Comment by anthony-swirldslabs** [comment]
> We currently don't have a good process for running these pre-release builds to verify these dashboard/metric name changes. Perhaps moving forward we could do before/after screenshots showing the changes?

There's indeed not a good process because the metric needs to exist in order to check if the dashboard plots it correctly. I suppose we'll be only able to test this in full once the code hits the previewnet and/or testnet.

**Review by jeromy-cannon — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Pull request merged**